### PR TITLE
SD-8429: Update documentation and explain versions

### PR DESCRIPTION
**Pull request by JureMalovrh**
Explain versions & update docs + re-generate

### CODE REVIEW

File: docs/resources/environment.md [hunk 1]
```diff
@@ -3,26 +3,26 @@
...
 subcategory: ""
 description: |-
-  Sleuth environment.
+  Project resource manages Sleuth project.
 ---
 
 # sleuth_environment (Resource)
 
-Sleuth environment.
+Project resource manages Sleuth project.
```
Comment (zigomir) [inline docs/resources/environment.md:11]: Should say environment here

File: docs/resources/incident_impact_source.md [hunk 2]
```diff
@@ -3,12 +3,12 @@
 page_title: "sleuth_incident_impact_source Resource - terraform-provider-sleuth"
 subcategory: ""
 description: |-
-  Sleuth incident impact source
+  Sleuth code change source.
```
Comment (zigomir) [inline docs/resources/incident_impact_source.md:6]: Should say incident impact here…

File: docs/resources/incident_impact_source.md [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
 ---
 
 # sleuth_incident_impact_source (Resource)
 
-Sleuth incident impact source
+Sleuth code change source.
```
Comment (zigomir) [inline docs/resources/incident_impact_source.md:11]: same as above

File: docs/resources/metric_impact_source.md [hunk 4]
```diff
@@ -47,20 +47,21 @@ resource "sleuth_metric_impact_source" "cloudwatch_rds_cpu" {
...
 - `name` (String) Impact source name
-- `project_slug` (String) The project for this environment
+- `project_slug` (String) The slug of the project that this metric impact source belongs to.
 - `provider_type` (String) Integration provider type
 - `query` (String) The metric query
 
 ### Optional
 
-- `integration_slug` (String) Integration slug is generated automatically when an integration is set up in Sleuth. By default, it matches the `provider_type`. Any value specified in the integration's `Description label` field gets appended to the `integration_slug`, spaces replaced with dashes, e.g. `cloudwatch-test`
+- `integration_slug` (String) The integration slug
```
Comment (zigomir) [inline docs/resources/metric_impact_source.md:58]: I guess we can omit internal implementation details we used to expose here...although it seems like that has been documented for a reason 🤔

**Review by zigomir — APPROVED** [review]
lgtm except for where resource names are off

File: docs/resources/metric_impact_source.md (hunk 4 above)
Comment (JureMalovrh) [reply docs/resources/metric_impact_source.md:58]: hm, it should remain, will check what caused the generate issue

**Review by zigomir — APPROVED** [review]

**Pull request merged**